Marlin/src/HAL/STM32/eeprom_flash.cpp

```diff
@@ -139,6 +142,11 @@ bool PersistentStore::access_start() {
 bool PersistentStore::access_finish() {
 
   if (eeprom_data_written) {
+    #ifdef STM32F4xx
+      // MCU may come up with flash error bits which prevent some flash operations.
+      // Clear flags prior to flash operations to prevent errors.
+      __HAL_FLASH_CLEAR_FLAG(FLASH_FLAG_OPERR | FLASH_FLAG_WRPERR | FLASH_FLAG_PGAERR | FLASH_FLAG_PGPERR | FLASH_FLAG_PGSERR);
```

Comment:
You know, what's interesting is the stm32_eeprom.c has a lot of different `__HAL_FLASH_CLEAR_FLAG` in their `#if defined` soup.

I think in `eeprom_buffer_flush` they are doing some clear errors but for the STM32F4xx they have nothing. Strange. I thought I had some error cleaning flags in the FLASH_EEPROM_LEVELING but I must have removed them before I submitted the PR.

Note: My references to the stm32 framework is to framework-arduinoststm32@3.10700.191028.

Reply:
@randellhodges I think the STM32F4 hits the `else` case and clears a couple flags, but not what I was seeing set.

Reply:
I edited my comment and didn't finish my thought sorry. "brain faster than keyboard"

I meant to say, it looks like they have a lot of flags clearing, but strangely enough for the F4, they go into an else and don't clear anything it, at first glance anyway.